### DISCUSSION TITLE
Pull devcontainer config from repo

### DISCUSSION THREAD

**Discussion by m4nch0t** [community/community#5909] [Codespaces]
Hi everyone,
I'm going crazy with git subtree or submodule to achieve my goal. Maybe someone can help me to find a way.

My problem is quite simple, I want to create a git project where my devcontainer configuration (devcontainer.json, Dockerfile, etc..) are stored and versioned. We always work on the same container custom image (for Ansible precisely) and I don't want to copy/paste my devcontainer configuration in all our projects (50+). And when we made a little modification in our Dockerfile or devcontainer.json, I don't want to modify and commit all my projects. Could be scripted but not very sexy after all.
I'm not particularly a git expert, so how can I achieve this goal?

Thank you in advance for helping.

Édit: 

I would like to recreate the same thing as an include in gitlab-ci.yml

include:
  remote: 'https://gitlab.com/awesome-project/raw/main/.before-script-template.yml'

**Comment by tanmayeekamath** [comment]
Hey @m4nch0t Thanks for your feedback. We have plans to add support for a private dev container registry / marketplace that could be used to maintain company created assets (container images); orgs could reference definitions from this marketplace by an identifier in their project’s devcontainer.json file. Any changes that happen to the image would then automatically apply to all the devcontainer configs referencing it. 

Would love to hear your feedback around this :)

**Reply by m4nch0t** [reply]
Hello @tanmayeekamath 
That's look awesome, I'm really impatient to test this !
